### DISCUSSION TITLE
Invalid argument error when calling in Perl, happens 10% of time

### DISCUSSION THREAD

**Discussion by kqe** [containers/podman#23731] [Q&A]
Hi, I call Podman, rootless, as a system() call in Perl script, which runs by a non-root user inside the email server to process mails.  

In Perl I set up the following env variables:
```
XDG_RUNTIME_DIR
HOME
```

I also had to create a dozen SELinux modules, so that it works on all the few servers.

But when trying on another server, it works well for 90% of the mails it processed.  For the other 10%, it issued exit code 1 with following error (issued through Perl):

cannot setns: Invalid argument

I couldn't google any similar error.  What I can do?

It's the stock version Podman on Redhat 7, still supported by RedHat extended, not sure if too old.

**Comment by Luap99** [comment]
> It's the stock version Podman on Redhat 7, still supported by RedHat extended, not sure if too old.

We only support the latest release upstream

**Comment by rhatdan** [comment]
Please upgrade to a RHEL8 or RHEL9 if possible.
